service/volume.py

```diff
@@ -8,10 +6,13 @@
 from core.models.identity import Identity
 
 from service.cache import get_cached_driver
-from service.driver import _retrieve_source
+from service.driver import _retrieve_source, prepare_driver
 
+from service import exceptions
 from service.instance import boot_volume_instance
-from service.exceptions import OverQuotaError
+
+# FIXME: fix prepare_driver to take a user directly
+Request = namedtuple("request", ["user"])
```

Comment:
I do like the name of `prepare_driver` I'm going to be working on moving Credentials into a permission class and authorize a persons credentials before creating a driver. These changes could simplify the driver code.